src/Components/ResultList.tsx

```diff
@@ -59,7 +59,7 @@ export default class ResultList extends React.Component {
         {this.state.results.map((result: Result) => {
           const template = this.headlessResultTemplateManager.selectTemplate(
             result
-          );
+          ) as any;
```

Comment:
Instead of adding `as any` here, it would be better to fix the generic type on line 31.
On line 35, it looks like content is a function that accepts a result and returns JSX (which is typed as `any`). We could pass this info directly, but a more readable way would be to store it in a type. So,

```
type Template = (result: Result) => any;
this.headlessResultTemplateManager = new ResultTemplatesManager<Template>(headlessEngine);
```